Bio/PDB/StructureAlignment.py

```diff
@@ -78,6 +107,49 @@ def __init__(self, fasta_align, m1, m2, si=0, sj=1):
         self.map21 = map21
         self.duos = duos
 
+    def _generate_alignment_from_models(self, m1, m2):
+        """Generate a MultipleSeqAlignment from two protein models .
+
+        Uses Bio.Align.PairwiseAligner to create a proper sequence alignment
+        with gaps, rather than simply placing sequences side-by-side.
+        This ensures that homologous residues are properly aligned.
+        """
+        seq1_record = self._extract_sequence_from_model(m1, "structure1")
+        seq2_record = self._extract_sequence_from_model(m2, "structure2")
+
+        aligner = PairwiseAligner("blastp")
+
+        alignments = aligner.align(seq1_record.seq, seq2_record.seq)
+        best_alignment = alignments[0]
+
+        aligned_seq1_str = str(best_alignment[0])
+        aligned_seq2_str = str(best_alignment[1])
+
+        # Convert to SeqRecord objects for MultipleSeqAlignment
+        aligned_seq1 = SeqRecord(Seq(aligned_seq1_str), id="structure1")
+        aligned_seq2 = SeqRecord(Seq(aligned_seq2_str), id="structure2")
+
+        records = [aligned_seq1, aligned_seq2]
+        alignment = MultipleSeqAlignment(records)
+        return alignment
+
+    def _extract_sequence_from_model(self, model, seq_id):
+        """Extract amino acid sequence from a protein model."""
+        residues = Selection.unfold_entities(model, "R")
+        sequence = ""
+
+        for residue in residues:
+            if isinstance(residue, Residue) and is_aa(residue):
+                resname = residue.get_resname()
+                if resname in PDBData.protein_letters_3to1_extended:
```

Comment:
Since `PDBData.protein_letters_3to1_extended` is a dictionary, you can do this instead:

```
resname = residue.get_resname()
aa_code = PDBData.protein_letters_3to1_extended.get(resname, "X")
sequence += aa_code
```